pkg/logql/engine.go

```diff
@@ -240,8 +240,15 @@ func (q *query) Exec(ctx context.Context) (logqlmodel.Result, error) {
 }
 
 func (q *query) Eval(ctx context.Context) (promql_parser.Value, error) {
-	userID, _ := tenant.TenantID(ctx)
-	queryTimeout := q.limits.QueryTimeout(userID)
+	// set as timeout the longest timeout across all tenants.
```

Comment:
So, we usually take the most restrictive limit. That'd be the smallest timeout. What is the reason to take the longest here?

Reply:
To avoid a smaller tenant messing up with other tenants. Full answer on https://github.com/grafana/loki/pull/7708#issuecomment-1322287483.